### PR TITLE
Move from Docker Hub to Quay.io

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Task

### Description

This PR implements the move from Docker Hub to Quay.io to avoid the pull limits. This was proposed in https://github.com/strimzi/proposals/blob/master/014-move-docker-images-to-quay.io.md

This PR changes the build and release tooling as well as the configurations of the Azure and Jenkins pipelines.

### Checklist

- [x] Update documentation
- [x] Update CHANGELOG.md

### CODE REVIEW

**Comment by scholzj** [comment]
@strimzi-ci run tests

**Comment by scholzj** [comment]
The actual push to Quay.io which uses the changed functionality is not used in PRs (which do not push anything anywhere) but only on the master branch or release tags. Bu the CIs seem to work the errors in Jenkins are unrelated to my changes. So this should be ready for review.

**Review by Frawless — APPROVED** [review]

File: helm-charts/helm2/strimzi-kafka-operator/README.md [hunk 1]
```diff
@@ -84,49 +84,63 @@ the documentation for more details.
...
+| `image.registry`                     | Cluster Operator image registry           | `quay.io`                                            |
 | `image.repository`                   | Cluster Operator image repository         | `strimzi`                                            |
 | `image.name`                         | Cluster Operator image name               | `cluster-operator`                                   |
 | `image.tag`                          | Cluster Operator image tag                | `latest`                                             |
-| `image.imagePullPolicy`              | Image pull policy for all pods deployed by Cluster Operator       | `IfNotPresent`                                       |
-| `image.imagePullSecrets`              | Docker registry pull secret              | `nil`                                                 |
+| `image.imagePullPolicy`              | Image pull policy for all pods deployed by Cluster Operator       | `IfNotPresent`               |
+| `image.imagePullSecrets`             | Docker registry pull secret               | `nil`                                                |
```
Comment (ppatierno) [inline helm-charts/helm2/strimzi-kafka-operator/README.md:92]: just "Image registry"? no "Docker" anymore? wdyt?

Comment (scholzj) [reply helm-charts/helm2/strimzi-kafka-operator/README.md:92]: I really just fixed the table alignment there. There are many places where we use Docker registry. So if you want to change that, I suggest to do that in separate PR.

**Review by ppatierno — APPROVED** [review]

**Pull request merged**